xarray/core/rolling.py

```diff
@@ -200,11 +208,61 @@ def _get_keep_attrs(self, keep_attrs):
 
         return keep_attrs
 
+    def _get_rolling_dim_coords(self, all_dims=False) -> Dict[str, Any]:
```

Comment:
I like the suggestion to call it `_get_output_coords`.  I'll make that change.

Regarding `all_dims`, in some cases, the return value is used in a select statement.  In that case, we only care about the dimensions where the coordinates are different than `self.obj.coords`, such as when there is no padding (this might be the only case).

In other cases, the return value is used in a call to `DataArray` or `Dataset`.  In these cases, we assume that the data has already been reduced to the proper size along each dimension (if applicable), and we wish to construct something akin to the current `DataArray` or `Dataset`, so we need/want all of the dimensions.

I've added a comment to try to clarify this.

Reply:
I just realized that I've been mixing up "dims" and "coordinates".  The `all_dims` argument should be `all_coords`, and should return the coordinates used to construct the rolling window if `False`, and all coordinates if `True`.

I'll fix the name.  Hopefully it makes more sense now.